Source/Scene/Material.js

```diff
@@ -853,13 +854,29 @@ function createTexture2DUpdateFunction(uniformId) {
       return;
     }
 
+    if (uniformChanged && defined(texture)) {
+      if (
+        texture !== this.defaultTexture &&
+        texture !== context.defaultTexture
+      ) {
+        texture.destroy();
```

Comment:
I'm noticing a momentary blip where the old texture is no longer used and the new texture is still loading in. Previously the swap was smooth. Is this fixable?

Reply:
I tried to take care of that by moving the `destroy()` call to the `prototype.update` function -- it would only `destroy` once the new texture was transferred from the `loadedImages` array. I'm not sure how to fix this in the scope of this update function because the new texture is handled by `prototype.update`, not this one?

Reply:
I see. I was thinking about an alternative like destroying the texture in the `.otherwise` callback, but that's not right either because creation/destruction of WebGL resources should happen in the render loop.

Maybe go back to how you originally had it but add a comment explaining why the code is there.